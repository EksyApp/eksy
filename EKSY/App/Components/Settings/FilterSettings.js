import React, {Component} from 'react'
import {View} from 'react-native'
import CheckBoxList from "../Common/CheckBoxList";
import Divider from "../Common/Divider";
import Label from "../Common/Label";

export default class FilterSettings extends Component {
	
	render() {
		return(
				<View>
					<Divider />
					<Label>Filters</Label>
					<CheckBoxList data={this.props.data} onPress={(name, checked) => {this.props.onPress(name, checked)}} />
				</View>
		)
	}
	
}