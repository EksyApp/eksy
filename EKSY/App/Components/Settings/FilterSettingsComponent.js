import React, {Component} from 'react'
import {View, StyleSheet} from 'react-native'
import CheckBoxList from "../Common/CheckBoxList";
import Divider from "../Common/Divider";
import Label from "../Common/Label";
import PropTypes from 'prop-types'
import {FiltersShape} from "../../Utils/PropTypeShapes";

export default class FilterSettingsComponent extends Component {
	
	render() {
		return(
				<View>
					<Divider style={styles.divider} />
					<Label>Filters</Label>
					<CheckBoxList data={this.props.data} onPress={(name, checked) => {this.props.onPress(name, checked)}} />
				</View>
		)
	}
	
}

const styles = StyleSheet.create({
	divider: {
		marginTop: 30
	}
})

FilterSettingsComponent.propTypes = {
	data: FiltersShape,
	onPress: PropTypes.func
}