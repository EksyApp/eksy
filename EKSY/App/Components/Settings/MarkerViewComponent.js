import React, {Component} from 'react'
import {ScrollView, View, StyleSheet} from 'react-native'
import Header from '../Common/Header'
import MarkerView from '../Common/MarkerView'
import Button from '../Common/Button'
import * as Theme from '../../Theme'
import PropTypes from 'prop-types'
import {MarkerShape, RegionShape} from '../../Utils/PropTypeShapes'
import Container from '../Common/Container'

export default class MarkerViewComponent extends Component {
	
	renderEditButton() {
		if (this.props.user && this.props.user.firebaseUser.uid === this.props.marker.creationInfo.user) {
			return (
					<Button onPress={this.props.onEditClick}>
						Edit
					</Button>
			)
		}
	}
	
	render() {
		return (
				<Container title="Marker" backButton>
					<MarkerView
							marker={this.props.marker}
							currentRegion={this.props.currentRegion}
					/>
					{this.renderEditButton()}
				</Container>
		)
	}
	
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: Theme.backgroundColor
	}
})

MarkerViewComponent.propTypes = {
	marker: MarkerShape,
	currentRegion: RegionShape,
	onEditClick: PropTypes.func
}