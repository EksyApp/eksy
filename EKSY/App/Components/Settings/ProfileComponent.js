import React, {Component} from 'react'
import Label from "../Common/Label";
import Button from "../Common/Button";
import {View} from "react-native";
import Message from "../Common/Message";
import {Actions} from 'react-native-router-flux'

export default class ProfileComponent extends Component {
	
	renderUserInfo() {
		if (this.props.user) {
			return (
					<Message>{this.props.user.firebaseUser.displayName}</Message>
			)
		}
	}
	
	renderProfileActionButton() {
		if (this.props.user) {
			return (
					<Button onPress={() => {this.props.onSignoutClick()}}>
						Signout
					</Button>
			)
		} else {
			return(
					<Button onPress={() => {Actions.login()}}>Login</Button>
			)
		}
	}
	
	renderShowMarkersButton() {
		if (this.props.user) {
			return (
					<Button onPress={() => {this.props.onMarkersClick()}}>
						Markers
					</Button>
			)
		}
	}
	
	render() {
		return (
				<View>
					<Label>Profile</Label>
					{this.renderUserInfo()}
					{this.renderShowMarkersButton()}
					{this.renderProfileActionButton()}
				</View>
		)
	}
	
	
}