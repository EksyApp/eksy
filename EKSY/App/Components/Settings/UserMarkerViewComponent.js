import React, {Component} from 'react'
import {ScrollView, View, StyleSheet} from "react-native";
import Header from "../Common/Header";
import MarkerView from "../Common/MarkerView";
import Button from "../Common/Button";
import * as Theme from "../../Theme";
import PropTypes from 'prop-types'
import {MarkerShape, RegionShape} from "../../Utils/PropTypeShapes";

export default class UserMarkerViewComponent extends Component {
	
	render() {
		return(
				<View style={styles.container}>
					<Header title="Your marker" backButton />
					<ScrollView style={styles.container}>
						<MarkerView
								marker={this.props.marker}
								currentRegion={this.props.currentRegion}
						/>
						<Button onPress={this.props.onEditClick}>
							Edit
						</Button>
					</ScrollView>
				
				</View>
		)
	}
	
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: Theme.backgroundColor
	}
})

UserMarkerViewComponent = {
	marker: MarkerShape,
	currentRegion: RegionShape,
	onEditClick: PropTypes.func
}