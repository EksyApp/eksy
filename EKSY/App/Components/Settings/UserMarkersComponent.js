import React, {Component} from 'react'
import {Text, View, StyleSheet} from "react-native";
import Header from "../Common/Header";
import * as Theme from "../../Theme";
import Message from '../Common/Message'
import MarkerCardList from '../Common/MarkerCardList'
import Label from "../Common/Label";
import Button from "../Common/Button";


export default class UserMarkersComponent extends Component {
	
	
	
	renderList() {
		if(this.props.loading) {
			return(<Message>Loading</Message>)
		} else {
			return(<MarkerCardList data={this.props.userMarkers} onPress={this.props.onCardClick} style={styles.card} />)
		}
	}
	
	render () {
		return (
				<View style={styles.container}>
					<Header title='Your markers' backButton />
					<Button onPress={this.props.onRefresh}>
						Refersh
					</Button>
					{this.renderList()}
				</View>
		)
	}
	
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: Theme.backgroundColor,
	},
	
	card: {
		width: '95%',
		alignSelf: 'center'
	}
})
