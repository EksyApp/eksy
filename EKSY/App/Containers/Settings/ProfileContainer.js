import React, {Component} from 'react'
import ProfileComponent from "../../Components/Settings/ProfileComponent";
import {connect} from "react-redux";
import firebase from 'firebase'
import * as ReduxActions from "../../Actions";
import {Actions} from 'react-native-router-flux'
import PropTypes from 'prop-types'
import {UserShape} from "../../Utils/PropTypeShapes";

export class ProfileContainer extends Component {
	
	async signout() {
		await firebase.auth().signOut()
		this.props.userSignedOut()
	}
	
	render() {
		console.log(this)
		return (
				<ProfileComponent
						user={this.props.user}
						onSignoutClick={() => {
							this.signout()
						}}
						onMarkersClick = {() => Actions.usersMarkers()}
				/>
		)
	}
	
}

ProfileContainer.propTypes  = {
	userSignedOut: PropTypes.func,
	user: UserShape
}

const mapStateToProps = (state) => {
	return {
		user: state.auth.user
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		userSignedOut: () => {dispatch(ReduxActions.userSignedOut())}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(ProfileContainer)