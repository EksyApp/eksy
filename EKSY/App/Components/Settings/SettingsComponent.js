import React, {Component} from 'react'
import { View, StyleSheet } from "react-native";
import Header from "../Common/Header";
import FilterSettingsContainer from "../../Containers/Settings/FilterSettingsContainer";
import * as Theme from "../../Theme";
import ProfileContainer from "../../Containers/Settings/ProfileContainer";


export default class SettingsComponent extends Component {
	
	render () {
		return (
				<View style={styles.container}>
					<Header title='Settings' backButton />
					<ProfileContainer/>
					<FilterSettingsContainer/>
				</View>
		)
	}
	
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: Theme.backgroundColor,
	}
})