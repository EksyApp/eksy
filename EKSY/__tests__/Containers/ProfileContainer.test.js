import React from 'react'
import { ProfileContainer } from '../../App/Containers/Settings/ProfileContainer'
import renderer from 'react-test-renderer'

test('ProfileContainer renders correctly', () => {
	const tree = renderer.create(
			<ProfileContainer
					userSignedOut={() => {}}
					user={{firebaseUser: {}}}
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});