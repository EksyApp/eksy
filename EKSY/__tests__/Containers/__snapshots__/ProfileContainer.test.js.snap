// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileContainer renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgb(101,114,122)",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Profile
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgb(89,85,78)",
          "fontSize": 18,
          "textAlign": "center",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "rgb(230,232,227)",
          "borderColor": "rgb(101,114,122)",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 15,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "rgb(101,114,122)",
            "fontSize": 16,
            "fontWeight": "600",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        Markers
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "rgb(230,232,227)",
          "borderColor": "rgb(101,114,122)",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 15,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "rgb(101,114,122)",
            "fontSize": 16,
            "fontWeight": "600",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        Routes
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "rgb(230,232,227)",
        "borderColor": "rgb(101,114,122)",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 15,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "color": "rgb(101,114,122)",
          "fontSize": 16,
          "fontWeight": "600",
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      Signout
    </Text>
  </View>
</View>
`;
